### PR TITLE
Publish/coverage mutation result listener

### PR DESCRIPTION
**Pull request by nikomall34**
You can ignore this branch. I just created it to publish and rename the coverage-plugin.

### CODE REVIEW

File: coverage-reporter/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+plugins {
+    kotlin("jvm") version "1.9.21"
+    `maven-publish`
+    idea
+    `java-gradle-plugin`
+    id("signing")
+}
+
+group = "io.github.amosproj"
```
Comment (lheimbs) [inline coverage-reporter/build.gradle.kts:12]: can we align this with the override-plugin?
so as id something like
`io.github.amos-pitmutationmate.pitmutationmate.coverage-reporter`

File: coverage-reporter/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+    repositories {
+        maven {
+            credentials {
+                username = project.findProperty("spaceUsername").toString()
+                password = project.findProperty("spacePassword").toString()
+            }
+            url = uri("https://maven.pkg.jetbrains.space/pitmutationmate/p/main/coverage-reporter")
+        }
+    }
```
Comment (lheimbs) [inline coverage-reporter/build.gradle.kts:56]: [This](https://www.jetbrains.com/help/space/consume-maven-artifacts.html) in step 5 reads like one would have to always specify the repository creds even as a user?
Or am i understanding this wrong?
It sounds like a private repository

**Comment by lheimbs** [comment]
Also why use this branch only for publishing? I think these are valid changes and can be reused if we/someone else want to publish an update

File: coverage-reporter/build.gradle.kts (hunk 2 above)
Comment (nikomall34) [reply coverage-reporter/build.gradle.kts:56]: Yes exactly. So the way I implemented here is that you need the credentials stored in the gradle.properties file, which is not uploaded to the repo.

File: coverage-reporter/build.gradle.kts (hunk 1 above)
Comment (nikomall34) [reply coverage-reporter/build.gradle.kts:12]: Yes I can change that. I am just struggling with uploading the plugin at all. After I did that I will try to change it.

**Comment by nikomall34** [comment]
Yes that's true. Once I am done I would merge it to main.

File: coverage-reporter/build.gradle.kts (hunk 2 above)
Comment (lheimbs) [reply coverage-reporter/build.gradle.kts:56]: Yeah i get that but the way I understand the documentation of this repository is that it is entirely **private** as in i could only download the coverage listener if i had the credentials. So no one can use this without them. Not only publish. Thats what it sounds like to me at least from the documentation

Comment (nikomall34) [reply coverage-reporter/build.gradle.kts:56]: Ah okay I see your point. Actually I have to publish it here https://s01.oss.sonatype.org. I also changed that locally. Once I can use the coverage-reporter through maven-central I will upload my changes.

**Comment by nikomall34** [comment]
I get this message when trying to upload the coverage-reporter with the new groupid:
Invalid groupId, com.github.* Group IDs are invalid now. The only allowed groupIds for Github projects are io.github.*
io.github.amosproj would be valid based on your Project URL.
To continue the registration process, please follow these steps:
Create a temporary, public repository called https://github.com/amosproj/OSSRH-98411 to verify github account ownership.
Edit this ticket, update the Group ID field with the new GroupId, and set Status to Open.
More info: https://central.sonatype.org/publish/requirements/coordinates/

As you can see in this case we have to take the groupID io.github.amosproj.

**Comment by lheimbs** [comment]
The error message sounds like you used `com.github...` is that the case? what about the one from the override plugin?

**Comment by nikomall34** [comment]
No, I used the GroupID you suggested `io.github.amos-pitmutationmate.pitmutationmate`. To publish to maven central, you have to create a sonatype-jira ticket. [Here](https://issues.sonatype.org/browse/OSSRH-98411?filter=-2) is the one I created. If you cannot access it, I can give you the credentials.

As far as I understand the [coordinates](https://central.sonatype.org/publish/requirements/coordinates/) the groupID is linked to the domain. Since the github domain (with the username) is `https://github.com/amosproj`, the groupID must start with `io.github.amosproj`. I could use the subgroup `io.github.amosproj.pitmutationmate` but `io.github.amosproj` must be present in the beginning.

I assume the `github.com` in the error message was mentioned because this is an autogenerated message. So if there is an error with groupID and your project is on github.com, the bot will always mention it.

**Review by lheimbs — APPROVED** [review]
> I assume the `github.com` in the error message was mentioned because this is an autogenerated message. So if there is an error with groupID and your project is on github.com, the bot will always mention it.

Yeah I figured as much. So for me it's fine to keep it as is.

**Pull request merged**